### PR TITLE
Add support of timestamps in nanoseconds

### LINKED ISSUE #364: Add support of timestamps in nanoseconds
The new built in function `from_timestamp` of the `time` module doesn't support nanoseconds which is inconsistent with other functions of the module and too limited. The goal of this ticket is to add the ability to convert timestamps in nanoseconds into date time.

### PR DESCRIPTION
**Pull request by essobedo**
fixes #364 

## Motivation

For many real life use cases, it is important to be a able to support timestamps expressed in nanoseconds.

## Modifications:

* Overloads the function `from_timestamp` to be able to set nanoseconds.
* Fixes the documentation of the built in function `from_timestamp` 
* Improves the documentation (not related to this issue)
* Adds a test for the function `now` since it was not covered by any tests (not related to this issue)

### CODE REVIEW

**Comment by essobedo** [comment]
Hi @adonovan and @brandjon WDYT of this one?

**Comment by essobedo** [comment]
Please note that I intentionally avoided to overload the function `from_timestamp` for clarity and simplicity

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:103]: A timestamp (aka UNIX time_t) is defined as the number of seconds since Jan 1, 1970, which may be an integer, or a fixed-point number (pair of int64) if sub-second precision is needed. It would be confusing for this function to introduce the new concept of a "timestamp" that is a quantity of nanoseconds.

Instead, change the existing from_timestamp function so that it accepts either one int (a number of seconds) or two ints (sec and nsec). It could additionally accept one float, a number of seconds. There is no need for a new function.

Also, the functions provided by this module should be documented at `Module`. I realize this is a preexisting problem, but would you mind addressing it while you're working here? (Look at json and math for examples.)

**Review by adonovan — CHANGES_REQUESTED** [review]

File: lib/time/time.go
Comment (essobedo) [reply lib/time/time.go]: @adonovan I understand your point of view even if it is not really helpful in my use case since I have timestamps expressed in nanoseconds so I will need to explicitly split seconds from nanoseconds which is not really convenient. Anyway I addressed your remarks.

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:22]: "nano" (throughout)

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:21]: s/amount/number/

Use "number" for countable nouns, and "amount" for mass nouns (e.g. amount of time).

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:23]: This line can be deleted if you make the previous line read

"number of seconds and (optionally) nanoseconds"

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:25]: The function should return an error, not an invalid value, if the argument is out of range.

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:28]: "is_valid_timezone(loc) - Reports whether loc is a valid time zone name."

(And delete the second sentence.)

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:31]: Add: Applications may replace this function by a deterministic one.

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:34]: The adjective "signed" applies to numbers, not character strings. Remove "possibly signed" and instead add:

A "-" prefix indicates a negative duration.

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:40]: Give an example of RFC 3339 format, e.g. "2021-03-22T23:20:50.52Z"

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:41]: "optional, "

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:50]: Indent each line by at least three spaces so that it appears in godoc as a block quote.

File: lib/time/time.go
Comment (adonovan) [inline lib/time/time.go:38]: parse_time

The design of this function is quite Go-specific, and I wonder whether it was a good idea to expose it directly. In any case, it needs more documentation about how the format string is interpreted. For now, let's redirect readers to Go's time.Parse function, since that is the actual behavior, though this is not ideal for a Starlark spec.

**Review by adonovan — COMMENTED** [review]
Thanks!

File: lib/time/time.go
Comment (essobedo) [reply lib/time/time.go:22]: Oops, good catch

**Review by adonovan — APPROVED** [review]
@brandjon LGTM, ready for you to merge.

**Review by ssoroka — APPROVED** [review]

**Comment by essobedo** [comment]
@brandjon Is-this PR good enough to be merged?

**Comment by essobedo** [comment]
@brandjon Could you please at least tell me if this project won't be maintained anymore? Because I wanted to work on https://github.com/google/starlark-go/issues/241 during my spare time, once this PR will be merged, if this project is no more supported/maintained I won't do it. Thank you in advance for your answer 🙏

**Comment by brandjon** [comment]
Apologies for letting this slip. Alan has write access to this repo and I've confirmed with him that he can merge.

**Pull request merged**
